packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -166,7 +190,109 @@ class RenderAndroidView extends RenderBox {
 
   @override
   void handleEvent(PointerEvent event, HitTestEntry entry) {
-    _motionEventsDispatcher.handlePointerEvent(event);
+    if (event is PointerDownEvent) {
+      _gestureRecognizer.addPointer(event);
+    }
+  }
+
+  @override
+  void detach() {
+    _gestureRecognizer.reset();
+    super.detach();
+  }
+}
+
+class _AndroidViewGestureRecognizer extends OneSequenceGestureRecognizer {
+  _AndroidViewGestureRecognizer(this.dispatcher, List<OneSequenceGestureRecognizer> gestureRecognizers) {
+    this.gestureRecognizers = gestureRecognizers;
+  }
+
+  final _MotionEventsDispatcher dispatcher;
+
+  // Maps a pointer to a list of its cached pointer events.
+  // Before the arena for a pointer is resolved all events are cached here, if we win the arena
+  // the cached events are dispatched to the view, if we lose the arena we clear the cache for
+  // the pointer.
+  final Map<int, List<PointerEvent>> cachedEvents = <int, List<PointerEvent>> {};
+
+  // Pointer for which we have already one the arena, events for pointers in this set are
+  // immediately dispatched to the Android view.
+  final Set<int> forwardedPointers = new Set<int>();
```

Comment:
Wasn't sure why some fields are 'public' and some have _ prefix. Made me think for a sec that this wasn't internal but seems like it is.